pkg/operator/options/options_validation.go

```diff
@@ -31,11 +32,20 @@ func (o Options) Validate() error {
 		o.validateRequiredFields(),
 		o.validateEndpoint(),
 		o.validateVMMemoryOverheadPercent(),
+		o.validateNetworkPluginMode(),
 		o.validateVnetSubnetID(),
 		validate.Struct(o),
 	)
 }
 
+func (o Options) validateNetworkPluginMode() error {
+	// NOTE: Network Plugin Mode should be normalized on the AKS API Level, so if we are passing values in from the ManagedCluster, NetworkPluginMode will already be normalized.
+	if o.NetworkPluginMode != consts.PodNetworkTypeOverlay && o.NetworkPluginMode != "" && o.NetworkPluginMode != "none" {
```

Comment:
"none" isn't on most of all clusters in the datamodel I could part with not including it.

Reply:
fair enough.